packages/jellyfish-api-core/src/category/masternode.ts

```diff
@@ -0,0 +1,146 @@
+import { ApiClient } from '../.'
+
+export enum MasternodeState {
+  PRE_ENABLED = 'PRE_ENABLED',
+  ENABLED = 'ENABLED',
+  PRE_RESIGNED = 'PRE_RESIGNED',
+  RESIGNED = 'RESIGNED',
+  PRE_BANNED = 'PRE_BANNED',
+  BANNED = 'BANNED',
+  UNKNOWN = 'UNKNOWN'
+}
+
+/**
+ * Masternode RPCs for DeFi Blockchain
+ */
+export class Masternode {
+  private readonly client: ApiClient
+
+  constructor (client: ApiClient) {
+    this.client = client
+  }
+
+  /**
+   * Creates a masternode creation transaction with given owner and operator addresses.
+   *
+   * @param {string} ownerAddress Any valid address for keeping collateral amount
+   * @param {string} [operatorAddress]  Masternode operator auth address (P2PKH only, unique). If empty, owner address will be used.
+   * @param {CreateMasternodeOptions} [options]
+   * @param {UTXO[]} [options.utxos = []]
+   * @param {string} [options.uxtos.txid] The transaction id
+   * @param {string} [options.utxos.vout] The output number
+   * @return {Promise<string>}
+   */
+  async createMasternode (
+    ownerAddress: string,
+    operatorAddress?: string,
+    options: CreateMasternodeOptions = { utxos: [] }
+  ): Promise<string> {
+    operatorAddress = operatorAddress ?? ownerAddress
+    return await this.client.call('createmasternode', [ownerAddress, operatorAddress, options.utxos], 'number')
+  }
+
+  /**
+   * Returns information about multiple masternodes.
+   *
+   * @param {MasternodePagination} pagination
+   * @param {string} [pagination.start]
+   * @param {boolean} [pagination.including_start = true] Include starting position.
+   * @param {string} [pagination.limit = 100] Maximum number of orders to return.
+   * @param {boolean} [verbose = true] Flag for verbose list. Only ids are returned when false.
+   * @return {Promise<MasternodeResult>}
+   */
+  listMasternodes (pagination?: MasternodePagination, verbose?: boolean): Promise<MasternodesResult<MasternodeInfo>>
```

Comment:
```suggestion
  listMasternodes (pagination?: MasternodePagination, verbose?: boolean): Promise<MasternodeResult<string | MasternodeInfo>>
```

Reply:
@fuxingloh , is there any specific reason for making the return type `Promise<MasternodeResult<string | MasternodeInfo>>`? The way i see it, `listMasternodes` will never return a string even if verbose is set to `false`

Reply:
I don't understand what you mean

1.`await client.masternode.listMasternodes({}, false)` -> `return Record<string, string>`
2.`await client.masternode.listMasternodes({}, true)` -> `return Record<string, Masternode>`

But I think this interface is redundant and you can remove as you will never hit this overload scenario. Since you already declare the other scenarios.

Reply:
Yes, all two scenarios above are correct.